scripts/codegen.ts

```diff
@@ -52,117 +49,138 @@ const getInputType = (input: ABIInputItem) => {
     }
 };
 
-async function main(folders: string[], subDir?: string) {
-    folders.forEach(async (folder) => {
-        const pascalName = kebabToPascalCase(folder).split('.')[0];
-        const constantName = kebabToConstantCase(folder).split('.')[0];
-
-        const remainingPath = 'dist/' + (subDir ? `${subDir}/` : '') + folder;
-        const dirName = process.cwd() + '/' + remainingPath;
-
-        const dirs = (await fs.readdir(dirName, { withFileTypes: true }))
-            .filter((dirent: any) => dirent.isDirectory())
-            .map((dirent: any) => dirent.name);
-
-        if (dirs.length > 0) {
-            await main(dirs, subDir ? subDir + '/' + folder : folder);
-        } else {
-            const { stdout: abiFileJson } = await $`cat ${remainingPath}/${pascalName}.json`;
-            const folderName = folder.split('.')[0];
-
-            const GENERATED_DISCLAIMER = `
-                /* eslint-disable @typescript-eslint/no-explicit-any */
-                /**
-                 * This file was generated by scripts/codegen.ts
-                 * 
-                 * Original abi file: 
-                 * - ${remainingPath}/${pascalName}.json
-                 * 
-                 * DO NOT EDIT MANUALLY
-                 */
-                `;
+/**
+ * Represents the contract folders to be codegen'd under artifacts/contracts
+ */
+const CONTRACT_FOLDERS = ['InterchainTokenService.sol', 'InterchainTokenFactory.sol', 'interchain-token', 'token-manager'];
```

Comment:
Can you include everything under contracts folder?

Reply:
what else needs to go in there? are all of those interactable?

Reply:
there's a lot of stuff that doesn't seem necessary for consumers under the artifacts/contracts folder, like `test`,`proxies`,`interfaces`,`utils`

Reply:
Utils are relevant for things transferring mint/burn permission. Any reason interfaces should be skipped? Other things can be relevant for custom tokens. I would put in everything except test since it might not just be us using it. What's the package size with that?